data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexManager.java

```diff
@@ -5,4 +5,5 @@
 public interface IndexManager{
 	void setupIndex() throws IOException;
 	String getIndexName(final String indexAlias) throws IOException;
+	boolean checkISMEnabled() throws IOException;
```

Comment:
Now that we are looking to expose this, it may make more sense to create a new method such as `isIndexAlias`. It could internally hold a boolean and thus avoid unnecessary checks on the cluster. As it is now, by calling this method twice, Data Prepper will unnecessarily make two requests to the cluster to get cluster settings.

Also, I believe this method (`checkISMEnabled`) is actually checking only the cluster. I believe the actual decision is a combination of two things: 1) The index is configured as an alias, and 2) the cluster supports ISM. e.g `currentIndexUsesAlias && clusterHasIsmEnabled`.

Reply:
I made some changes to the logic to meet your comment. I am not sure, whether ISM really needs to be enabled, though. Please have a look at the new version. It also includes unit tests for `isIndexAlias()`.